### PR TITLE
Close file handle after use in generated code

### PR DESCRIPTION
**Pull request by jonboulle**
h/t @runcom

### CODE REVIEW

**Comment by runcom** [comment]
:+1:

**Comment by jonboulle** [comment]
took the liberty of fixing travis too

File: main.go [hunk 1]
```diff
@@ -348,6 +348,7 @@ func {{.FunctionPrefix}}FSByte(useLocal bool, name string) ([]byte, error) {
 		if err != nil {
 			return nil, err
 		}
+		defer f.Close()
```
Comment (mjibson) [inline main.go]: No need for this to be a defer. Save the results of ReadAll, close, then return.

Comment (runcom) [inline main.go]: isn't it the same but w/o having to store the result of ReadAll and check for the error? meh though it's less code like this

Comment (runcom) [inline main.go]: also you need to make sure to Close on ReadAll failure as well no?

Comment (runcom) [inline main.go]:
```
b, err := ioutil.ReadAll(f)
if err != nil {
    f.Close()
   return nil, err
}
f.Close()
return b, nil
```

Comment (mjibson) [inline main.go]: Defer is a slow operation, and so shouldn't be used if there's only one return path. There's no difference in the return values if ReadAll returns an error, so the extra logic you have above doesn't add any value.

Comment (runcom) [inline main.go]: I don't understand, if readall fails you still have to close the file

Comment (mjibson) [inline main.go]:
```
b, err := ioutil.ReadAll(f)
f.Close()
return b, err
```

Comment (runcom) [inline main.go]: Ok i see nvm

**Comment by jonboulle** [comment]
@mjibson updated

**Pull request merged**